internal/pkg/manifest/workload_test.go

```diff
@@ -389,6 +389,60 @@ func TestPlacementArgOrString_UnmarshalYAML(t *testing.T) {
 	}
 }
 
+func TestSubnetListOrArgs_UnmarshalYAML(t *testing.T) {
+	testCases := map[string]struct {
+		inContent []byte
+
+		wantedStruct SubnetListOrArgs
+		wantedError  error
+	}{
+		"returns error if both string slice and args specified": {
+			inContent: []byte(`subnets: ["id1", "id2"]
+  from_tags:
+    - foo: bar`),
+
+			wantedError: errors.New("yaml: line 1: did not find expected key"),
+		},
+		"error if unmarshalable": {
+			inContent: []byte(`subnets:
+  ohess: linus
+  archie: leg64`),
+			wantedError: errUnmarshalSubnetsOpts,
+		},
+		"success with string slice": {
+			inContent: []byte(`subnets: ["id1", "id2"]`),
+			wantedStruct: SubnetListOrArgs{
+				IDs: []string{"id1", "id2"},
+			},
+		},
+		"success with args": {
+			inContent: []byte(`subnets:
+  from_tags:
+    foo: bar`),
```

Comment:
Just trying to understand the realtime scenario here. What will be actual values if not `foo: bar`?
Will it be `subnetName: subnetId`?

Reply:
Okay I got it. I checked the `FilterForTags` method and understood the use case